### PR TITLE
home-assistant: 2021.6.6 -> 2021.7.0

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

https://www.home-assistant.io/blog/2021/07/07/release-20217/

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/octoprint/default.nix [hunk 1]
```diff
@@ -50,19 +50,25 @@ let
...
+                inherit (oldAttrs) pname;
                 inherit version;
                 sha256 = "0jpgd0rk91si93857mjrizan5gc42kj1q4fi4160qgk68la88fl9";
               };
-              buildInputs = [ self.nose ];
-              checkPhase = "nosetests";
+              pythonImportsCheck = [
+                "zeroconf"
+              ];
+              buildInputs = with self; [
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/octoprint/default.nix:67]: You are sure this shouldn't be checkInputs?

**Comment by mweinelt** [comment]
As it is, lots of failing tests. Lots of issues due to failed multicast membership requests (`[19] No such device`) when component tests try to set up zeroconf, ssdp or upnp. This is messy, no clear path in my mind on how to sidestep this.

```
=========================== short test summary info ============================
FAILED tests/components/default_config/test_init.py::test_setup[pyloop] - ass...
FAILED tests/components/homekit/test_homekit.py::test_homekit_uses_system_zeroconf[pyloop]
FAILED tests/components/vultr/test_init.py::TestVultr::test_setup - Attribute...
FAILED tests/components/vultr/test_sensor.py::TestVultrSensorSetup::test_invalid_sensors
FAILED tests/components/vultr/test_sensor.py::TestVultrSensorSetup::test_sensor
FAILED tests/components/vultr/test_binary_sensor.py::TestVultrBinarySensorSetup::test_binary_sensor
FAILED tests/components/vultr/test_binary_sensor.py::TestVultrBinarySensorSetup::test_invalid_sensors
FAILED tests/components/vultr/test_switch.py::TestVultrSwitchSetup::test_invalid_switches
FAILED tests/components/vultr/test_switch.py::TestVultrSwitchSetup::test_switch
FAILED tests/components/vultr/test_switch.py::TestVultrSwitchSetup::test_turn_off
FAILED tests/components/vultr/test_switch.py::TestVultrSwitchSetup::test_turn_on
FAILED tests/components/zeroconf/test_usage.py::test_multiple_zeroconf_instances[pyloop]
FAILED tests/components/zeroconf/test_usage.py::test_multiple_zeroconf_instances_gives_shared[pyloop]
FAILED tests/components/zeroconf/test_init.py::test_setup[pyloop] - assert False
FAILED tests/components/zeroconf/test_init.py::test_setup_with_overly_long_url_and_name[pyloop]
FAILED tests/components/zeroconf/test_init.py::test_setup_with_default_interface[pyloop]
FAILED tests/components/zeroconf/test_init.py::test_setup_without_default_interface[pyloop]
FAILED tests/components/zeroconf/test_init.py::test_setup_without_ipv6[pyloop]
FAILED tests/components/zeroconf/test_init.py::test_setup_with_ipv6[pyloop]
FAILED tests/components/zeroconf/test_init.py::test_setup_with_ipv6_default[pyloop]
FAILED tests/components/zeroconf/test_init.py::test_zeroconf_match_macaddress[pyloop]
FAILED tests/components/zeroconf/test_init.py::test_zeroconf_match_manufacturer[pyloop]
FAILED tests/components/zeroconf/test_init.py::test_zeroconf_match_manufacturer_not_present[pyloop]
FAILED tests/components/zeroconf/test_init.py::test_zeroconf_no_match[pyloop]
FAILED tests/components/zeroconf/test_init.py::test_zeroconf_no_match_manufacturer[pyloop]
FAILED tests/components/zeroconf/test_init.py::test_homekit_match_partial_space[pyloop]
FAILED tests/components/zeroconf/test_init.py::test_homekit_match_partial_dash[pyloop]
FAILED tests/components/zeroconf/test_init.py::test_homekit_match_partial_fnmatch[pyloop]
FAILED tests/components/zeroconf/test_init.py::test_homekit_match_full[pyloop]
FAILED tests/components/zeroconf/test_init.py::test_homekit_already_paired[pyloop]
FAILED tests/components/zeroconf/test_init.py::test_homekit_invalid_paring_status[pyloop]
FAILED tests/components/zeroconf/test_init.py::test_homekit_not_paired[pyloop]
FAILED tests/components/zeroconf/test_init.py::test_get_instance[pyloop] - as...
FAILED tests/components/zeroconf/test_init.py::test_removed_ignored[pyloop]
FAILED tests/components/zha/test_discover.py::test_devices[zha_device_joined-pyloop-device98]
FAILED tests/components/zha/test_discover.py::test_devices[zha_device_restored-pyloop-device98]
FAILED tests/components/zha/test_discover.py::test_discover_endpoint[pyloop-device_info98]
ERROR tests/components/fritz/test_config_flow.py
ERROR tests/components/default_config/test_init.py::test_setup[pyloop] - OSEr...
ERROR tests/components/sonos/test_init.py::test_creating_entry_sets_up_media_player[pyloop]
ERROR tests/components/sonos/test_media_player.py::test_async_setup_entry_hosts[pyloop]
ERROR tests/components/sonos/test_plex_playback.py::test_plex_play_media[pyloop]
ERROR tests/components/sonos/test_switch.py::test_entity_registry[pyloop] - O...
ERROR tests/components/sonos/test_init.py::test_configuring_sonos_creates_entry[pyloop]
ERROR tests/components/sonos/test_sensor.py::test_entity_registry_unsupported[pyloop]
ERROR tests/components/sonos/test_media_player.py::test_async_setup_entry_discover[pyloop]
ERROR tests/components/sonos/test_switch.py::test_alarm_attributes[pyloop] - ...
ERROR tests/components/sonos/test_init.py::test_not_configuring_sonos_not_creates_entry[pyloop]
ERROR tests/components/sonos/test_sensor.py::test_entity_registry_supported[pyloop]
ERROR tests/components/sonos/test_media_player.py::test_services[pyloop] - OS...
ERROR tests/components/sonos/test_switch.py::test_alarm_create_delete[pyloop]
ERROR tests/components/sonos/test_media_player.py::test_device_registry[pyloop]
ERROR tests/components/sonos/test_sensor.py::test_battery_attributes[pyloop]
ERROR tests/components/sonos/test_media_player.py::test_entity_basic[pyloop]
ERROR tests/components/sonos/test_sensor.py::test_battery_on_S1[pyloop] - OSE...
ERROR tests/components/sonos/test_sensor.py::test_device_payload_without_battery[pyloop]
ERROR tests/components/upnp/test_config_flow.py::test_flow_ssdp_discovery[pyloop]
ERROR tests/components/upnp/test_init.py::test_async_setup_entry_default[pyloop]
ERROR tests/components/upnp/test_config_flow.py::test_flow_ssdp_incomplete_discovery[pyloop]
ERROR tests/components/upnp/test_config_flow.py::test_flow_ssdp_discovery_ignored[pyloop]
ERROR tests/components/upnp/test_config_flow.py::test_flow_user[pyloop] - OSE...
ERROR tests/components/upnp/test_config_flow.py::test_flow_import[pyloop] - O...
ERROR tests/components/upnp/test_config_flow.py::test_flow_import_already_configured[pyloop]
ERROR tests/components/upnp/test_config_flow.py::test_flow_import_no_devices_found[pyloop]
ERROR tests/components/upnp/test_config_flow.py::test_options_flow[pyloop] - ...
ERROR tests/components/zha/test_climate.py::test_set_moes_preset[pyloop] - At...
ERROR tests/components/zha/test_climate.py::test_set_moes_operation_mode[pyloop]
= 37 failed, 13691 passed, 34 skipped, 2 xfailed, 6768 warnings, 30 errors in 1174.59s (0:19:34) =
```

Vultr is basically failing like in #125962.

```
E           AttributeError: <class 'vultr.vultr.Vultr'> does not have the attribute 'server_list'
```

**Comment by fabaff** [comment]
Dependency updates for 2021.7.2: https://github.com/NixOS/nixpkgs/pull/130055

**Comment by mweinelt** [comment]
That's great, but we need to fix the tests first. I need to know why all these components are now trying to join a multicast group. If you could invest time to dig into the hass codebase that'd be great. I can't make heads or tails of it.

**Comment by mweinelt** [comment]
Disabled tests for ssdp, upnp, zeroconf and sonos. They all try to join multicast groups in the build sandbox, but fail to find a proper interface to do so.

That was a mess.

**Pull request merged**